app/javascript/packs/validateFileInputs.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * @file Manages logic to validate file uploads client-side. In general, the
+ * validations work by looping over input form fields with a type of file and
+ * checking the size and format of the files upload by the user.
+ */
+
+/**
+ * An object containing the top level MIME type as the key and the max file
+ * size in MB for the value. To use a different value than these defaults,
+ * simply add a data-max-file-mb attribute to the input form field with the
+ * max file size in MB. If that attribute is found, it takes priority over these
+ * defaults.
+ *
+ * @constant {Object.<string, number>}
+ */
+const MAX_FILE_SIZE_MB = Object.freeze({
+  image: 2,
```

Comment:
Should we set this default to 25mb as well?

Not sure what we want videos to be at but we can handle that later.

Reply:
Re images: I set it to 2 because that's what the [`ProfileImageUploader`](https://github.com/thepracticaldev/dev.to/blob/1b822c2d802208c301b617a78fecec3f476cddf4/app/uploaders/profile_image_uploader.rb#L7) has it set at and more than just the `User` model uses that uploader. But now that I'm thinking about it, it seems Article images are more likely the most common case so I'll bump it up. S3 is cheap 😎

Re videos: Agreed, I didn't implement this functionality with videos because they currently use [`s3_direct_upload`](https://github.com/thepracticaldev/dev.to/blob/1b822c2d802208c301b617a78fecec3f476cddf4/app/views/videos/new.html.erb#L15) which is another issue entirely 😂. I realized that after the fact and figured it didn't hurt to leave in.